### PR TITLE
1-2-3-4: Display the current and required Bitcoin confirmation count

### LINKED ISSUE #297: Capture unapplied mainnet tweaks
We've got a few tweaks that we applied on mainnet in #210 that didn't bubble back to the main branch. Let's make sure these tweaks, or the appropriate adaptations based on the current dApp, land on the main branch. Following is a list.

- [x] Handling mainnet config (https://github.com/keep-network/tbtc-dapp/commit/f1536a829bcd55f59f8fa4f94944d1c604c6dff9). Ideally, this would be a toggle we'd make based on the Ropsten chain id (see #205 for some initial work in this dept). -> see #352 
- [x] Clarifying that funding should not happen from an exchange (https://github.com/keep-network/tbtc-dapp/pull/210/commits/ff5c294e6917023d713390cfa9d3da58bf56bf0a). -> see #352 
- [x] Call out the exact nature of the expected BTC payment (https://github.com/keep-network/tbtc-dapp/commit/8dffb7d4d5608a1072e9e2f16a13b0ff9007f5d3). -> see #352 
- [ ] Clarify how many Bitcoin confirmations we are waiting for (https://github.com/keep-network/tbtc-dapp/commit/63dbad8750864ab13e333b35f625de768fef5c6a). This should live in [the `Confirming` component](https://github.com/keep-network/tbtc-dapp/blob/master/src/components/deposit/Confirming.js#L24). Ideally we would read the confirmation count from the constants contract, and track how many confirmations we've already observed, but that's likely best served by a separate issue.

Remaining commits on that branch have either been obsoleted or applied.

### PR DESCRIPTION
**Pull request by ironng**
[PR soundtrack 🎵 ](https://www.youtube.com/watch?v=ABYnqp-bxvg)

Clarifies how many Bitcoin confirmations we are waiting for and what the current count is for both the deposit and redemption flows.

Closes #297 

References tbtc.js changes in https://github.com/keep-network/tbtc.js/pull/80

### CODE REVIEW

**Comment by Shadowfiend** [comment]
I think we can say this closes 297 tbh. Will give this a test drive tomorrow!

File: src/sagas/deposit.js [hunk 1]
```diff
@@ -301,6 +304,50 @@ export function* getBitcoinAddress() {
...
+        emit(END)
+      }
+    }
+
+    deposit.onReceivedFundingConfirmation(listener)
+
+    // no-op
+    // eventChannel subscriber must return an unsubscribe, but tbtc.js does not
+    // currently provide one
+    return () => {}
```
Comment (Shadowfiend) [inline src/sagas/deposit.js:327]: What structure is this unsubscribe in? We can probably add this in future work.

File: src/sagas/deposit.js [hunk 2]
```diff
@@ -301,6 +304,50 @@ export function* getBitcoinAddress() {
...
+  })
+}
+
+function* watchForFundingConfirmations(deposit) {
+  yield delay(500)
```
Comment (Shadowfiend) [inline src/sagas/deposit.js:332]: What are these delays for? May be worth flagging with a comment as well.

File: src/sagas/deposit.js [hunk 3]
```diff
@@ -301,6 +304,50 @@ export function* getBitcoinAddress() {
...
+      yield delay(50)
+      if (confirmations) {
+        yield put({
+          type: BTC_TX_CONFIRMED,
+          payload: { btcConfirmedTxID: transactionID, confirmations },
+        })
+      }
+    }
+  } finally {
+    console.debug("And now, the watch has ended")
```
Comment (Shadowfiend) [inline src/sagas/deposit.js:347]: Is this because `take` will `throw` when the channel closes?

**Comment by Shadowfiend** [comment]
🔢😉

**Comment by mhluongo** [comment]
:musical_note: 1-2-3-4 tell me your Proof of Work :musical_note:

File: src/sagas/deposit.js (hunk 1 above)
Comment (ironng) [reply src/sagas/deposit.js:327]: The `eventChannel` would want something like:
```
    const unsubscribe = () => {
      deposit.removeReceivedConfirmationHandler( listener)
    }

    return unsubscribe
```

File: src/sagas/deposit.js (hunk 2 above)
Comment (ironng) [reply src/sagas/deposit.js:332]: The delays are an attempt at debouncing. Actually I think this first delay doesn't really make sense, so I removed it as part of #355. It's an old way of doing it and maybe unnecessary? Here's what the [docs](https://redux-saga.js.org/docs/recipes/) say.

File: src/sagas/deposit.js (hunk 3 above)
Comment (ironng) [reply src/sagas/deposit.js:347]: Tbh I was interested in watching for debug statement. 😅 We can just remove it - without it the saga should just end.

**Comment by ironng** [comment]
@Shadowfiend I see what you mean about duplicating code! Made an attempt to DRY this up in #355

**Pull request merged**